tex/generic/pgf/basiclayer/pgfcorearrows.code.tex

```diff
@@ -510,12 +510,13 @@
   fill/.code={%
     \def\pgf@temp{#1}%
     \ifx\pgf@temp\pgf@nonetext%
-      \pgfarrowsaddtooptions{\pgfarrowopentrue}
+      \pgfarrowsaddtooptions{\pgfarrowopentrue}%
+    \else\ifx\pgf@temp\pgfkeysnovalue@text%
+      \pgfarrowsaddtooptions{\pgfarrowopenfalse}%
```

Comment:
According to the newly added doc, "If the argument is left empty or omitted altogether, ...", empty value should be treated the same as no value.

The `/tikz/fill` is implemented by setting default value to empty and doing `\edef` to its value.
https://github.com/pgf-tikz/pgf/blob/44f8137449b34f62bc6371bd442ae5cc98f60f18/tex/generic/pgf/frontendlayer/tikz/tikz.code.tex#L507-L519

Reply:
Pity, applying suggestions on deleted lines is not supported, yet.

```tex
\pgfkeys{
  /pgf/arrow keys/.cd,
  fill/.code={%
    \edef\pgf@temp{#1}%
    \ifx\pgf@temp\pgf@nonetext%
      \pgfarrowsaddtooptions{\pgfarrowopentrue}%
    \else\ifx\pgf@temp\pgfutil@empty
      \pgfarrowsaddtooptions{\pgfarrowopenfalse}%
    \else
      \pgfarrowsaddtooptions{\pgfarrowopenfalse\def\pgf@arrows@fill@color{#1}}%
    \fi\fi
  },
  fill/.default=,
}
```

Reply:
That's a greate suggestion. I think in that case we can even keep the `/.value required`.

Reply:
If I get you right, you meant
```tex
\pgfkeys{
  /pgf/arrow keys/.cd,
  fill/.code={%
    \def\pgf@temp{#1}% back to \def or keep using \edef?
    \ifx\pgf@temp\pgf@nonetext
      \pgfarrowsaddtooptions{\pgfarrowopentrue}%
    \else\ifx\pgf@temp\pgfutil@empty
      \pgfarrowsaddtooptions{\pgfarrowopenfalse}%
    \else
      \pgfarrowsaddtooptions{\pgfarrowopenfalse\def\pgf@arrows@fill@color{#1}}%
    \fi\fi
  },
  fill/.value required,
}
```